### PR TITLE
QUARKUS-1414 Service Binding improvement test plan

### PR DESCRIPTION
**Pull request by mjurc**
* Adding a test plan for QUARKUS-1414.

### CODE REVIEW

File: QUARKUS-1414.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+applications.
+
+## Scope of the testing
+
+### Impact on test suites and testing automation
+* We need to ensure all the supported means of producing Quarkus deployments in OpenShift work with service binding
+  * Source S2I
+  * Docker build
+  * Binary S2I
+* Orchestration of OSD will have to be automated to ensure easy testing on OSD with RHOSAK service bindings. This, as of
```
Comment (rsvoboda) [inline QUARKUS-1414.md:15]: Pleas provide info why OSD is needed and we can't use OCP instances in our lab.

File: QUARKUS-1414.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  now, is done manually (see QUARKUS-1027).
```
Comment (rsvoboda) [inline QUARKUS-1414.md:16]: link to QUARKUS-1027 / make it clickable

File: QUARKUS-1414.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* New tests have to be implemented in Quarkus QE test suite, with applications utilizing the following service bindings:
```
Comment (rsvoboda) [inline QUARKUS-1414.md:17]: Are these services sorted by priority? If not pls sort them according the priority / ask for prioritisation on RFE

File: QUARKUS-1414.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  * RHOSAK
+  * Apicurio
+  * MongoDB
+  * MySQL
+  * PostgreSQL
+
```
Comment (rsvoboda) [inline QUARKUS-1414.md:23]: Do we need to cover all of the above services in RHBQ testing.

I would be concerned primarily about RHOSAK / Kafka.

File: QUARKUS-1414.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+### Impact on resources
+* A developer preview instance of RHOSAK will have used along with OpenShift Dedicated running in RHBQ QE AWS quota
+  during the testing.
+* Service binding operator will have to be installed in QE OpenShift clusters.
+* Spin-up of the test applications (bindable services need to be instantiated) and test phase in test suite should take
+  tens of minutes altogether.
+
```
Comment (rsvoboda) [inline QUARKUS-1414.md:30]: What would be the budget impact?

What is the rough time estimation for TD?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-1414.md (hunk 3 above)
Comment (mjurc) [reply QUARKUS-1414.md:17]: No, there's no priorities, and remember, this still is TP

File: QUARKUS-1414.md (hunk 4 above)
Comment (mjurc) [reply QUARKUS-1414.md:23]: The original messaging by PM was that the unit testing is sufficient, which makes me considerably uncomfortable - but considering this is TP, there's no priorities, probably

File: QUARKUS-1414.md (hunk 5 above)
Comment (mjurc) [reply QUARKUS-1414.md:30]: I'd say upwards of 5 man days, as I really have no clue actually

File: QUARKUS-1414.md (hunk 1 above)
Comment (mjurc) [reply QUARKUS-1414.md:15]: This is the venue we've explored in 2.2, according to docs at that time. There is a RHOAS operator that is installable on any cluster, but I'd have to commit more to seeing how that works.

File: QUARKUS-1414.md (hunk 3 above)
Comment (rsvoboda) [reply QUARKUS-1414.md:17]: +1

File: QUARKUS-1414.md (hunk 4 above)
Comment (rsvoboda) [reply QUARKUS-1414.md:23]: IMHO RHOSAK is the main one, because it's priority for Middleware

File: QUARKUS-1414.md (hunk 5 above)
Comment (rsvoboda) [reply QUARKUS-1414.md:30]: so 5 MD as optimistic estimate, 10 MD as pessimistic estimate?

Please add info about effort estimation into TP.

Comment (mjurc) [reply QUARKUS-1414.md:34]: Updated.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**